pandas/core/common.py

```diff
@@ -1403,6 +1403,10 @@ def _fill_zeros(result, x, y, name, fill):
                 if np.isinf(fill):
                     np.putmask(result,(signs<0) & mask, -fill)
 
+                if "floordiv" in name:  # (PR 9308)
+                    nan_mask = ((y == 0) & (x == 0)).ravel()
+                    np.putmask(result, nan_mask, np.nan)
+
```

Comment:
I guess it's taking so long because my DataFrame is 10e8 floats (~1GB).  

Profiling the `_fill_zeros` method (without the if statement), it spends 0.36s on the `nan_mask = ` line (and  `nan_mask` uses 94MB).  It spends 0.06s on the following line.